picard/tagger.py

```diff
@@ -842,6 +842,66 @@ def finished(file, result):
+        """Extract AcousticBrainz features and submit them."""
+
+        def abz_extractor_callback(file, result, error):
+            file.clear_pending()
+
+            from concurrent.futures import Future
+            abz_metadata_file, result, error = result.result() if isinstance(result, Future) else result
+            if "Writing results" in error and abz_metadata_file:
+                file.acousticbrainz_features_file = abz_metadata_file
+                log.debug("AcousticBrainz extracted features of recording %s: %s" %
+                          (file.metadata["musicbrainz_recordingid"], file.filename,))
+
+                # Submit results
+                from picard.acousticbrainz import submit_features
+                submit_features(self, file, abz_metadata_file)
+            elif 'Duplicate' in error:
+                log.debug("AcousticBrainz already has an entry for recording %s: %s" %
+                          (file.metadata["musicbrainz_recordingid"], file.filename,))
+                file.acousticbrainz_is_duplicate = True
+            else:
+                # Something went wrong
+                log.debug("AcousticBrainz extraction failed with error: %s" % error)
+
+                if result == 1:
+                    log.warning("AcousticBrainz extraction failed: %s" % file.filename)
+                elif result == 2:
+                    log.warning("AcousticBrainz extraction failed due to missing a MusicBrainz Recording ID: %s" % file.filename)
+                else:
+                    log.warning("AcousticBrainz extraction failed due to an unknown error: %s" % file.filename)
+
+        for file in iter_files_from_objects(objs):
+            # Skip unmatched files
+            if not file.metadata["musicbrainz_recordingid"]:
+                log.warning("File does not have a MusicBrainz Recording ID: %s" % file.filename)
+            # And process matched ones
+            else:
+                file.set_pending()
+
+                from picard.acousticbrainz import run_extractor_entry
+                from tempfile import NamedTemporaryFile
+
+                # Create a temporary file with acousticbrainz output
+                output_file = NamedTemporaryFile("w", suffix=".json")
+                output_file.close()  # close file to ensure other processes can write to it
+
+                # Check if file was either already processed or sent to the AcousticBrainz server
+                if file.acousticbrainz_features_file:
+                    abz_extractor_callback(file, (file.acousticbrainz_features_file, 0, "Writing results"), False)
+                elif file.acousticbrainz_is_duplicate:
+                    abz_extractor_callback(file, (None, 0, "Duplicate"), False)
+                else:
+                    # Launch the acousticbrainz on a separate process
+                    log.debug("Extracting AcousticBrainz features from %s" % file.filename)
+                    run_extractor_entry(self,
```

Comment:
Same here, we usually prefer:

```python
verylongfunc(
    p1,
    p2,
    ....
)
```

Over:

```python
verylongfunc(p1,
             p2,
             ...)
```

Main reasons:
- it is easier to re-indent
- change in function name only affects one line
- change to a parameter only affects one line
- more space to write long parameter